File: HighlandsCore/src/main/java/highlands/highlandscore/HighlandsCore.java
package highlands.highlandscore;


import kr.entree.spigradle.annotations.SpigotPlugin;

import highlands.highlandscore.api.HighlandsCoreApi;
import highlands.highlandscore.commands.HighlandsCommandHandler;
import highlands.highlandscore.infrastructure.HighlandsCoreDB;
import highlands.highlandscore.settings.Settings;
import highlands.highlandscore.utilities.AdvancedLogger;
import org.bukkit.plugin.java.JavaPlugin;

import java.sql.SQLException;
import java.util.logging.Level;

@SpigotPlugin
public class HighlandsCore extends JavaPlugin {
    //Settings
    private Settings settings;
    public Settings getSettings() {
        return settings;
    }

    //Loggers
    private AdvancedLogger advancedLogger;
    public AdvancedLogger getAdvancedLogger() {
        return advancedLogger;
    }

    //Database
    private HighlandsCoreDB highlandsCoreDB;
    public HighlandsCoreDB getPlayerDataDB() {
        return highlandsCoreDB;
    }

    private HighlandsCoreApi highlandsCoreApi;
    public HighlandsCoreApi getHighlandsCoreApi() { return highlandsCoreApi; }

    @Override
    public void onEnable() {
        //Settings
        saveDefaultConfig();
        settings = new Settings(this);

        //Loggers
        advancedLogger = new AdvancedLogger(this);

        //DB
        highlandsCoreDB = new HighlandsCoreDB();
        if (settings.getUseMysql())
            highlandsCoreDB.setupMysql(this);
        else
            highlandsCoreDB.setupSQLite(this);
        try {
            highlandsCoreDB.connect();
        } catch (SQLException e) {
            advancedLogger.Log(Level.SEVERE, "HighlandsCore", e.getMessage());
        }

        //Api
        highlandsCoreApi = new HighlandsCoreApi(this);

        //Command
        registerCommands();
    }

    @Override
    public void onDisable() {
        try {
            highlandsCoreDB.disconnect();
        } catch (SQLException e) {
            advancedLogger.Log(Level.SEVERE, "HighlandsCore", e.getMessage());
        }
    }

    public void reloadPluginConfig() {
        this.reloadConfig();
    }

    private void registerCommands() {
        this.getCommand("hlc").setExecutor(new HighlandsCommandHandler());
    }
}

File: HighlandsCore/src/main/kotlin/com/highlands/highlandscore/HighlandsCore.kt
package com.highlands.highlandscore

import com.highlands.highlandscore.api.HighlandsCoreApi
import com.highlands.highlandscore.commands.HighlandsCommandHandler
import com.highlands.highlandscore.infrastructure.HighlandsCoreDB
import com.highlands.highlandscore.settings.Settings
import com.highlands.highlandscore.utilities.AdvancedLogger
import kr.entree.spigradle.annotations.PluginMain
import org.bukkit.plugin.java.JavaPlugin
import java.sql.SQLException
import java.util.logging.Level

@PluginMain
class HighlandsCore : JavaPlugin() {

    //Settings
    var settings: Settings? = null
        private set

    //Loggers
    var advancedLogger: AdvancedLogger? = null
        private set

    //Database
    var playerDataDB: HighlandsCoreDB? = null
        private set
    var highlandsCoreApi: HighlandsCoreApi? = null
        private set

    override fun onEnable() {
        //Settings
        saveDefaultConfig()
        settings = Settings(this)

        //Loggers
        advancedLogger = AdvancedLogger(this)

        //DB
        playerDataDB = HighlandsCoreDB()
        if (settings!!.useMysql) playerDataDB!!.setupMysql(this) else playerDataDB!!.setupSQLite(this)
        try {
            playerDataDB!!.connect()
        } catch (e: SQLException) {
            advancedLogger!!.Log(Level.SEVERE, "HighlandsCore", e.message)
        }

        //Api
        highlandsCoreApi = HighlandsCoreApi(this)

        //Command
        registerCommands()
    }

    override fun onDisable() {
        try {
            playerDataDB!!.disconnect()
        } catch (e: SQLException) {
            advancedLogger!!.Log(Level.SEVERE, "HighlandsCore", e.message)
        }
    }

    fun reloadPluginConfig() {
        reloadConfig()
    }

    private fun registerCommands() {
        getCommand("hlc")!!.setExecutor(HighlandsCommandHandler())
    }
}